### ISSUE TITLE
Critical bug in Realm 5.3.1, instances returned by Realm (by using findFirst(), findAll(), etc.) change to null

### ISSUE THREAD

**Issue by m-Bilal** [realm/realm-java#6056]
Okay, so my guess is that this one bug is causing a lot of unexpected behaviour in Realm 5.3.1. The following is what I observed:
1) realm.createObject(<Class Name>.class, int id) saves all values except for the id as null or 0. (realm.copyToRealm() works.)
2) Any query (such as findAll(), findFrist(), etc) returns an object(s) with the correct id but all other fields as null.
3) When using a static method to return RealmResults or an Object, the object fields have their correct values inside the function, but change to null once you try using them anywhere else.
Eg:
```Java
    public static RealmResults<SubscriptionTypeModel> getAllSubscriptionTypeModelsFromRealm() {
        Realm realm = Realm.getDefaultInstance();
        RealmResults<SubscriptionTypeModel> realmResults =
                        realm.where(SubscriptionTypeModel.class).findAll();
        for (SubscriptionTypeModel i: realmResults) {
            Log.d(TAG,"getAllSubscriptionModelsFromRealm, id: " + i.id);
            Log.d(TAG,"getAllSubscriptionModelsFromRealm, name: " + i.name);
            Log.d(TAG,"getAllSubscriptionModelsFromRealm, minage: " + i.minAge);
            Log.d(TAG,"getAllSubscriptionModelsFromRealm, maxage: " + i.maxAge);
        }
        return realmResults;
    }
```
The logs in this function will print the correct values. However if you log the returned results in another function of another class, all objects will have the correct id, but all other fields will be either null or 0.

NOTE: To anyone who is facing the same issues, you can downgrade your Realm version to 5.0.1 and Realm will start working fine.

**Comment by Zhuinden** [comment]
I'm guessing you use direct field access instead of getters/setters?

Just checking

**Comment by m-Bilal** [comment]
Yes

**Comment by nhachicha** [comment]
@m-Bilal 
So basically calling `getAllSubscriptionTypeModelsFromRealm` in one class/function works (i.e print the correct values) whereas calling it in another class/method doesn't?

If that's the case, the only thing I suspect is that the byte code changes are not applied correctly for some classes, causing it to return default values. This might be related to the incremental build change introduced recently...

Can you please check what happen if you:
1- Use a clean build? (you still experience the issue?)
2- Using explicitly the getter/setter

**Comment by m-Bilal** [comment]
Using a clean build didn't solve anything, but using getter/setter methods did. It works perfectly fine with getter/setter methods. Thanks for the advice!!

**Comment by nhachicha** [comment]
@m-Bilal we still need to investigate why this happens, is it possible for you to share your project? (privately if you want help@realm.io)

**Comment by m-Bilal** [comment]
@nhachicha 
I was able to recreate the error in a much more simple project so that you won't have to read tons of code.
Here's the link: https://github.com/m-Bilal/RealmErrorRecreate

**Comment by m-Bilal** [comment]
@nhachicha 
Okay, I'll force the full build. Thanks a lot for helping me out! Cheers!!

**Comment by cmelchior** [comment]
Fixed in #6058
